esp-config/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "std")]
+mod generate;
+#[cfg(feature = "std")]
+pub use generate::*;
+
+#[macro_export]
+macro_rules! esp_config_int {
+    ($ty:ty, $var:expr) => {
+        match <$ty>::from_str_radix(env!($var), 10) {
```

Comment:
Can we instead generate a `consts.rs` for numbers/strings?

Reply:
It would also allow us to keep esp-config a build-dependency.

Reply:
I believe we can actually already do this with just pure env variables as we _always_ output the variable to rustc: https://github.com/MabezDev/esp-hal/blob/poc/native-config/esp-config/src/generate.rs#L102

Therefore I think we can do the "front end" parsing in build.rs and normalize to decimal for the end application.

I need to test that's possible though.

Reply:
Unfortunately, `env!` always expands to a `&'static str`

Reply:
Yes but I think we can do a conversion in build.rs to allow the const parsing code to always get decimal values.

"0xAA" -> build.rs parses 170 -> updates the env string to be "170"

Reply:
And with that, we still need from_str_radix, or do you want to roll our own?

Reply:
I've implemented support for parsing hex, octal and binary and if all else fails it'll try and parse as a decimal. I've also switched to the hand rolled const int parsing (with a note that we could ditch it for the official API in the future.

With this in place, IMO there is no reason to have a file (we can still generate one for a human to read as per your other comment).

Reply:
Works for me, thanks